beacon_node/execution_layer/src/engine_api/http.rs

```diff
@@ -1147,14 +1168,41 @@ impl HttpJsonRpc {
     pub async fn forkchoice_updated(
         &self,
         forkchoice_state: ForkchoiceState,
-        payload_attributes: Option<PayloadAttributes>,
+        maybe_payload_attributes: Option<PayloadAttributes>,
     ) -> Result<ForkchoiceUpdatedResponse, Error> {
         let engine_capabilities = self.get_engine_capabilities(None).await?;
-        if engine_capabilities.forkchoice_updated_v2 {
-            self.forkchoice_updated_v2(forkchoice_state, payload_attributes)
+        if let Some(payload_attributes) = maybe_payload_attributes.as_ref() {
+            match payload_attributes {
+                PayloadAttributes::V1(_) | PayloadAttributes::V2(_) => {
+                    if engine_capabilities.forkchoice_updated_v2 {
+                        self.forkchoice_updated_v2(forkchoice_state, maybe_payload_attributes)
+                            .await
+                    } else if engine_capabilities.forkchoice_updated_v1 {
+                        self.forkchoice_updated_v1(forkchoice_state, maybe_payload_attributes)
+                            .await
+                    } else {
+                        Err(Error::RequiredMethodUnsupported("engine_forkchoiceUpdated"))
+                    }
+                }
+                PayloadAttributes::V3(_) => {
+                    if engine_capabilities.forkchoice_updated_v3 {
+                        self.forkchoice_updated_v3(forkchoice_state, maybe_payload_attributes)
+                            .await
+                    } else {
+                        Err(Error::RequiredMethodUnsupported(
+                            "engine_forkchoiceUpdatedV3",
+                        ))
+                    }
+                }
+            }
+        } else if engine_capabilities.forkchoice_updated_v3 {
+            self.forkchoice_updated_v2(forkchoice_state, maybe_payload_attributes)
```

Comment:
Yes! Nice catch. I must have gotten it backwards when I wrote it unless it was a typo.

You see once I change this to call `forkchoice_updated_v3`, this will result in lighthouse calling `forkchoice_updated_v3` as soon as the EL says it supports it, which should be **before the `deneb` fork**. Note the [specification](https://github.com/ethereum/execution-apis/blob/main/src/engine/cancun.md#specification-1) says this is an error condition **if we have a `PayloadAttributes`** but says nothing about this when we don't have one. After speaking with some EL devs it turns out to be easier for both of us if we agree that it's **not** an error condition to call `forkchoice_updated_v3` before the `deneb` fork when there is no `PayloadAttributes`.